rust: tests: cleanup to reduce flakiness

LeoComandini opened the pull request:
With the recent `cargo test` changes,
now tests are run with higher parallelism,
which make them even more flaky.
This MR attempts to fix some of the issues that
arose in some of the past CI runs.

jgriffiths commented on subprojects/gdk_rust/tests/integration.rs:
```diff
@@ -317,18 +317,42 @@ fn subaccounts(is_liquid: bool) {
-    // Allow some time for the new session to catch up
-    std::thread::sleep(std::time::Duration::from_secs(2));
+    // Wait until all subaccounts have been recovered
+    let mut i = 120;
```
nit: 500ms in a test doesn't matter, this would be clearer if it was 60, the sleep was 1 sec, and the '1 minute' comment just said 'timeout out waiting for updates' since the delay time may be tweaked in the future.

LeoComandini replied:
Updated in 69cd5b47d426b17e482a402f250c09499107ec61.

On a dedicated machine (which only builds and runs) it increases the test time by ~5s (up to 99s); but that's negligible since all integration tests are not run that often, and on busy boxes it's reasonable to expect that it will be even more negligible.

jgriffiths commented on subprojects/gdk_rust/tests/test_session.rs:
```diff
@@ -812,25 +785,24 @@ impl TestSession {
+        let height = initial_height_electrs + 1;
+        // Wait until electrs has updated
         let mut i = 120;
-        let new_height = loop {
```
as above

jgriffiths reviewed (commented):
utack 270cb457bda33105fc1a81972e754354a7fe8149 with (very minor) nit

LeoComandini commented:
> utack [270cb45](https://github.com/Blockstream/gdk/commit/270cb457bda33105fc1a81972e754354a7fe8149) with (very minor) nit

Addressed and added an unrelated commit if you want to review again

jgriffiths commented:
utack 69cd5b47d426b17e482a402f250c09499107ec61
